File: backend/app/routers/ai.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.services import ai_service
from app.database import get_db
from app.schemas.ai import AIRequest, AIResponse

router = APIRouter(prefix="/ai", tags=["AI"])


@router.post("/suggest", response_model=AIResponse)
def suggest(request: AIRequest, db: Session = Depends(get_db)):
    return ai_service.suggest(request, db)

File: frontend/src/services/ai.ts
import api from "./api"

export interface AIRequest {
    title: string;
    type: string;
}

export interface AIResponse {
    description: string;
    tags: string[];
}

export const suggest = async (payload: AIRequest) => {
    const { data } = await api.post<AIResponse>('/ai/suggest', payload);
    return data;
}
